util/eventbus/eventbustest/doc.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Package eventbustest provides helper methods for testing the eventbus.
+//
+// An event bus connects publishers of typed events with subscribers
+// interested in those events. Typically, there is one global event
+// bus per process.
+//
+// # Usage
+//
+// The test helper presents a set of generic ideas for testing and a
+// few specific helpers.
+//
+// Generally, tests can be run as:
+//
+//	bus := eventbus.New()
+//	defer bus.Close()
+//
+//	tw := bus.Debugger().NewTestWatcher()
+//	defer tw.Done()
+//
+//	somethingThatEmitsSomeEvent()
+//	if err := eventbus.Expect[EventFoo](tw); err != nil {
+//	  t.Error(err.Error())
+//	}
+//
+// This example is also the given example for the simples helper that
+// expects a particular event to have happened on the bus.
+//
+// The helpers are:
+//
+// Expect: Expect a given event of type
+//
+//	eventbustest.Expect[T any](tw *eventbustest.TestWatcher) error
+//
+// ExpectFunc: Expect a given even of type, with func to check event values.
+// Expect is implemented using ExpectFunc and a func of `{ return true }`.
+//
+//	eventbustest.ExpectFunc[T any](tw *eventbustest.TestWatcher, func(event T) (bool, error)) error
+//
+// ExpectEvents: Expect a given set of events allowing other events interspersed
+//
+//	eventbustest.ExpectEvents(tw *eventbustest.TestWatcher, events ...any) error
+//
+// ExpectOnlyEvents: Expect a given set of events but now allowing other events interspersed
+//
+//	eventbustest.ExpectOnlyEvents(tw *eventbustest.TestWatcher, events ...any) error
+//
+// The last two helpers uses the following struct as a tuple:
```

Comment:
I suggest folding these two lines into the next block, e.g.,
```suggestion
//
// As shown, [Expect] checks that at least one event of the given type occurs
// in the stream generated by the code under test.
//
// To check for specific properties of an event, use [ExpectFunc], which checks
// that at least one event of the given type occurs in the stream for which the
// given function reports true.
//
// To test for multiple events, use [ExpectEvents], which checks that the stream
// contains the given events in the given order, possibly with other events
// interspersed.
//
// To test the complete contents of the stream, use [ExpectOnlyEvents], which checks
// that the stream contains exactly the given events in the given order, and no
// others.
//
// The last two helpers uses the following struct as a tuple:
```

And so on below.

As noted here, I would also suggest _not_ duplicating the function signatures in the comment, since the doc tool will render them below (e.g., see https://godoc.org/tailscale.com/util/eventbus).